### ISSUE TITLE
Cloning visualization feature not working in scatter/data magic templates

### ISSUE THREAD

**Issue by h-robertson** [kiln/flourish-sdk#83]
When you try and use the cloning viz feature in the SDK (e.g. adding the ID of [this scatter visualization](https://app.flourish.studio/visualisation/10376816/edit) when running the scatter template) it doesn't seem to work anymore.

I'm not 100% sure if it's the data magicness that's preventing this, but I also tried one in the Table template (also a data magic one) and it doesn't work there anymore either.

Here's the errors from the console in Scatter:
<img width="406" alt="image" src="https://user-images.githubusercontent.com/40994811/174296758-74585037-9fbc-4089-a254-c0ea73f0f5f2.png">

**Comment by daanlouter** [comment]
Interesting.. I wonder if this is because it doesn't send the data magic metadata object? Does that sound reasonable, @oampo?

**Comment by oampo** [comment]
Yep, it looks like both `getLiveApiOptionsObject` and `getPublishedJsonObject` in `server/entity/private/visualisation/serializer.js` need updating to also include the metadata.

**Comment by oampo** [comment]
I think @bobbysebolao did the SDK changes to add metadata, so probably has more context than me on how the everything work there.
